src/app/shared/links/links-list/links-list.component.ts

```diff
@@ -17,21 +17,80 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import {Component} from '@angular/core';
+import {ChangeDetectionStrategy, Component, EventEmitter, Input, OnChanges, Output, SimpleChanges} from '@angular/core';
+import {DocumentModel} from '../../../core/store/documents/document.model';
+import {Observable, combineLatest as observableCombineLatest} from 'rxjs';
+import {LinkTypeModel} from '../../../core/store/link-types/link-type.model';
+import {selectLinkTypesByDocumentId} from '../../../core/store/link-types/link-types.state';
+import {AppState} from '../../../core/store/app.state';
+import {Store} from '@ngrx/store';
+import {map, tap} from 'rxjs/operators';
+import {selectCollectionsDictionary} from '../../../core/store/collections/collections.state';
+import {CollectionModel} from '../../../core/store/collections/collection.model';
+import {DocumentsAction} from '../../../core/store/documents/documents.action';
 
 @Component({
   selector: 'links-list',
   templateUrl: './links-list.component.html',
-  styleUrls: ['./links-list.component.scss']
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class LinksListComponent {
+export class LinksListComponent implements OnChanges {
 
-  public links: string[] = [
-    'Name1',
-    'name2',
-    'Name3 ja ja lsdjfh aljsdkf aljsd fajsdf ajds flajsdhfla dsfl ajsdf ajsdhf lasd fladsjhf jadsf ajdfhask',
-  ];
+  @Input() public document: DocumentModel;
 
-  public selectedLink: string = 'Name1';
+  @Output() public select = new EventEmitter<{ collection: CollectionModel, document: DocumentModel }>();
+
+  public linkTypes$: Observable<LinkTypeModel[]>;
+  public activeLinkType: LinkTypeModel;
```

Comment:
I cannot select linked document in Detail perspective and I think it is caused by using this property together with `OnPush` change detection. This property should be observable as well so the template will react to the changes properly.

Reply:
No it works as expected, but you have to do double click to select linked document. In some other PR we will introduce some icon to select document.